mesonbuild/cmake/tracetargets.py

```diff
@@ -46,10 +69,28 @@ def resolve_cmake_trace_targets(target_name: str,
             continue
 
         if curr not in trace.targets:
+            curr_path = Path(curr)
             if reg_is_lib.match(curr):
                 res.libraries += [curr]
-            elif Path(curr).is_absolute() and Path(curr).exists():
-                res.libraries += [curr]
+            elif curr_path.is_absolute() and curr_path.exists():
+                if any([x.endswith('.framework') for x in curr_path.parts]):
+                    # Frameworks detected by CMake are passed as absolute paths
+                    # Split into -F/path/to/ and -framework name
+                    path_to_framework = []
+                    # Try to slice off the `Versions/X/name.tbd`
+                    for x in curr_path.parts:
+                        path_to_framework.append(x)
+                        if x.endswith('.framework'):
+                            break
+                    curr_path = Path(*path_to_framework)
+                    framework_path = curr_path.parent
+                    framework_name = curr_path.stem
+                    framework_include = _get_framework_include_path(curr_path)
+                    res.libraries += [f'-F{framework_path}', '-framework', framework_name]
+                    if framework_include:
+                        res.include_directories += [f'-I{framework_include}']
```

Comment:
This seems related to https://github.com/mesonbuild/meson/issues/10002.

Reply:
Good catch. I need to fix another issue regarding include flag order, which is very similar to #11123.

Reply:
@nirbheek it would be great if you could check this out too.